dds/DCPS/RTPS/Spdp.cpp

```diff
@@ -666,7 +666,16 @@ Spdp::handle_participant_data(DCPS::MessageId id,
         DCPS::LogGuid(guid_).c_str(), DCPS::LogGuid(guid).c_str(),
         pdata.leaseDuration.seconds, DCPS::LogAddr(from).c_str(), participants_.size()));
     }
-
+    ::CORBA::Long maxLeaseDuration = config_->max_lease_duration().to_dds_duration().sec;
```

Comment:
This block should be conditional on !from_sedp (which would be a secure update).